src/main/scala/chiseltest/legacy/backends/verilator/VerilatorExecutive.scala

```diff
@@ -48,6 +52,74 @@ object VerilatorExecutive extends BackendExecutive {
     val circuit = elaboratedAnno.collect { case x: ChiselCircuitAnnotation => x }.head.circuit
     val dut = getTopModule(circuit).asInstanceOf[T]
 
+    if (elaboratedAnno.contains(CachingAnnotation)) {
+      // Generate new high FIRRTL for caching
+      val highFirrtlPath = s"${targetDir}/${circuit.name}.hi.fir"
+      val oldHighFirrtlPath = s"${targetDir}/${circuit.name}.hi.fir.old"
+      s"mv ${highFirrtlPath} ${oldHighFirrtlPath}".!
+
+      (new ChiselStage).run(
+        elaboratedAnno :+ RunFirrtlTransformAnnotation(new HighFirrtlEmitter)
+      )
+
+      // Diff high FIRRTL and delete old file
+      val diffExit = s"diff ${highFirrtlPath} ${oldHighFirrtlPath}".!
+      s"rm -rf ${oldHighFirrtlPath}".!
+
+      // Generate new hash
+      val hashFile = new File(targetDir, "elaborationDetails.hash")
+
+      // Hash the deterministic elements of elaboratedAnno for comparison
+      // currently since this is guaranteed to be in Verilator and have the same test run dir, this should
+      // only catch if coverage or vcd flags change
+      val elaboratedAnnoHash = elaboratedAnno
+        .toSeq
+        .filter(anno => !List("chisel3.stage.ChiselCircuitAnnotation", "chisel3.stage.DesignAnnotation")
+          .contains(anno.getClass.toString.split(" ").last))
+        .map(anno => anno.hashCode)
+      val firrtlInfoHash = Seq(firrtl.BuildInfo.hashCode)
```

Comment:
This isn't really good enough since `hashCode` is not a cryptographic hash, it is only really meant to quickly discover that two things are _not_ equal, but it isn't good enough for equality.
The easiest approach here might be to serialize all annotations to a JSON string and then use a Java `MessageDigest` to calculate a sha256 hash.